src/common-messagelog/src/main/java/ee/ria/xroad/common/messagelog/archive/GPGOutputStream.java

```diff
@@ -66,14 +72,13 @@
     private boolean closed = false;
 
     /**
-     * Constructs a stream that pipes data to a gpg process encrypting and signing.
-     *
-     * @param gpgHome        GnuPG home directory containing the secret key for signing.
-     * @param output         Path to the output file, overwritten if present.
-     * @param encryptionKeys Zero or more encryption key files in PGP format (see gpg --recipient-file)
+     * Constructs a stream that pipes data to a gpg process for encrypting and signing.
+     * @param gpgHome GnuPG home directory containing the secret key for signing.
+     * @param output Path to the output file, overwritten if present.
+     * @param encryptionKeys Zero or more encryption (recipient) key identifiers
      * @throws IOException if setting up the gpg process fails
      */
-    GPGOutputStream(Path gpgHome, Path output, List<Path> encryptionKeys) throws IOException {
+    GPGOutputStream(Path gpgHome, Path output, Set<String> encryptionKeys) throws IOException {
         super(null);
```

Comment:
The superclass constructor is annotated with @NotNull so it feels dangerous to pass null there.

`FilterOutputStream(@NotNull OutputStream out)`

Reply:
The `@NotNull` is probably inferred by the IDE, it is not part of the source code; `null` is also explicitly permitted by the API documentation (https://docs.oracle.com/javase/8/docs/api/java/io/FilterOutputStream.html#FilterOutputStream-java.io.OutputStream-)
> out - the underlying output stream to be assigned to the field this.out for later use, or null if this instance is to be created without an underlying stream.

The output stream is eventually assigned at the end of the constructor.